dhall/src/Dhall.hs

```diff
@@ -1582,33 +1582,69 @@ instance Inject Int where
 
         declared = Integer
 
+{-| Injecting `Word` types gives `Natural`
+>>> embed inject (12 :: Word)
```

Comment:
Please add an empty line after the first line of each haddocks section. I'm pretty sure haddock needs that.